lib/normalize-option.js

```diff
@@ -0,0 +1,22 @@
+'use strict';
+
+var valueOrFunction = require('value-or-function');
+
+module.exports = function normalizeOption(types, defaultValue, value) {
```

Comment:
yeah, the only problem with having `value-or-function` not aware of types is that you could never have a `function` typed option.  Maybe it just needs to be expanded to support an array of types (and handles the common case of a singular type in the fast path).